src/RequestBuilder.php

```diff
@@ -145,22 +145,22 @@ private function getConfigsForUriTemplates($config)
      */
     private function constructBodyAndQueryParameters(Message $message, $config)
     {
-        $messageData = json_decode($message->serializeToJsonString(), true);
+        $messageDataJson = $message->serializeToJsonString();
 
         if ($config['body'] === '*') {
-            return [$messageData, []];
+            return [$messageDataJson, []];
         }
 
         $body = null;
         $queryParams = [];
-
+        $messageData = json_decode($messageDataJson, true);
         foreach ($messageData as $name => $value) {
             if (array_key_exists($name, $config['placeholders'])) {
                 continue;
             }
 
             if (Serializer::toSnakeCase($name) === $config['body']) {
-                $body = $value;
+                $body = $message->{"get$name"}()->serializeToJsonString();
```

Comment:
I did a little looking around and it appears we should always expect a message type for the body, but as a safety net do you think it would be worth it to check to make sure we are working with a `Message` instance instead of, for example, a `RepeatedField`?

Reply:
This is a good idea. Something like this?:

```php
if (($bodyMessage = $message->{"get$name"}) instanceof Message) {
    $body = $bodyMessage
} else {
    $body = $value
}
```

Reply:
Yeah, that looks nice. We will also need to make sure `$value` gets JSON encoded if it isn't an instance of `Message` though.

Reply:
PTAL